### PR TITLE
Focus a channel by joining it

### PR DESCRIPTION
**Pull request by realies**
#1155

### CODE REVIEW

**Comment by MaxLeiter** [comment]
Commit message could be a little more descriptive, maybe do a `git commit --amend`?

**Comment by dgw** [comment]
@MaxLeiter The source branch is `patch-1`, so based on that plus the default title "Update lounge.js", it was probably committed from Github's website. No `--amend`ing there, AFAIK…

File: client/js/lounge.js [hunk 1]
```diff
@@ -829,6 +829,20 @@ $(function() {
 			return;
 		}
 
+		if (text.substring(0, 5) === "/join") {
```
Comment (xPaw) [inline client/js/lounge.js]: I think this will fail if you send `/joinabc`

Comment (realies) [inline client/js/lounge.js]: Your point being?

Comment (realies) [inline client/js/lounge.js]: Updated. A cleaner and a more efficient solution might require refactoring the event handler function.

**Comment by YaManicKill** [comment]
I would quite like this to be pulled out of this funciton a bit, having all this code inline doesn't feel ideal. I'm thinking a better way would be to have modules for commands that we want to deal with client side, rather than server side.

**Comment by realies** [comment]
Perhaps the event handler for form submission should be checking if the first character is `/` after which checking if the command has anything to do with the client in a more structured manner.

**Comment by xPaw** [comment]
This should be fine, however we do need to have some proper logic for handling commands on the client, otherwise we will end up adding even more indexofs/substrings to check for commands there.

**Comment by realies** [comment]
@xPaw, agreed. Perhaps if the first character of the input is `/`, switch between a list of valid client-side handled functionalities. Thoughts?

**Comment by xPaw** [comment]
@realies That probably will work, have the function return true if it has been "eaten", and if the function does not return true, send it to the server as usual.

File: client/js/lounge.js [hunk 2]
```diff
@@ -829,6 +829,20 @@ $(function() {
 			return;
 		}
 
+		if (text.substring(0, 6) === "/join ") {
+			var cmd = text.trim();
```
Comment (MaxLeiter) [inline client/js/lounge.js]: `const`?

File: client/js/lounge.js [hunk 3]
```diff
@@ -829,6 +829,20 @@ $(function() {
...
+
+			if (cmd !== "/join") {
+				var name = cmd.split(" ").pop();
```
Comment (MaxLeiter) [inline client/js/lounge.js]: `const`?

File: client/js/lounge.js [hunk 4]
```diff
@@ -829,6 +829,20 @@ $(function() {
...
+				var chan = findCurrentNetworkChan(name);
```
Comment (MaxLeiter) [inline client/js/lounge.js]: `const`?

**Comment by astorije** [comment]
@realies, let us know when you can apply the few changes @xPaw and @MaxLeiter mentioned, it would be very nice if we could get this in :)

**Comment by realies** [comment]
@astorije, will do as soon as possible.

**Comment by astorije** [comment]
Hey @realies, friendly ping? :)

**Comment by xPaw** [comment]
Superseded by #1491.

**Pull request closed without merging**